wagtail/wagtailadmin/templatetags/wagtailadmin_tags.py

```diff
@@ -39,6 +40,17 @@ def main_nav(context):
         'request': request,
     }
 
+
+@register.inclusion_tag('wagtailadmin/shared/search_other.html', takes_context=True)
+def search_other(context):
```

Comment:
This tag could take a single argument for the currently active search type: `{% search_other current="pages" %}`:
```
@register.inclusion_tag('wagtailadmin/shared/search_other.html', takes_context=True)
def search_other(context, current=None):
    request = context['request']

    return {
        'options_html': admin_search_areas.render_html(request, current),
        'request': request,
    }
```

This could then be passed through when rendering [`SearchArea.render_html`](https://github.com/torchbox/wagtail/pull/1800/files#diff-b1c5845438a86b2864088e26ac56596aR98):
```
def render_html(self, request, current=None):
    # ...
    for item in sorted(search_areas, key=lambda i: i.order):
        rendered_search_areas.append(item.render_html(request, query, current=item.name == current))
    # ...
```

This way, matching the current item does not have to depend upon the URL, and the 'Other searches' bar could be shown on any page where it might be relevant, without losing the ability to mark a search type as current.